internal/services/stock_exchange/algorithms.go

```diff
@@ -0,0 +1,340 @@
+package stock_exchange
```

Comment:
why stock_exchange.go is deleted? can we keep the service methods there and write algo in this file?

Reply:
Will add that in a new branch/pr. Deleted it here as it was causing errors while writing the code.